.github/workflows/docker.yml

```diff
@@ -5,6 +5,8 @@ on:
   push:
     branches:
     - develop
+  release:
+    types: [published]
```

Comment:
I'm very rusty at dockerhub stuff at the moment, but IIRC, shouldn't we tag the "release" images with the `git tag` of the pybamm's release?

I mean - won't this overwrite the "latest" (develop) image on dockerhub with the "release" image, and then as soon as a new commit is pushed to develop, the "release" image will be overwritten by a new "latest" image?

Reply:
Yes, I'm aware. We're still discussing atm about how the tags will be set since we want to label the image with either the PyBaMM version or the name of the release (since it contains the version) – the latter is easier to extract from the GitHub events API. Marking the PR as draft for now since it's not ready yet

Reply:
> shouldn't we tag the "release" images with the `git tag` of the pybamm's release?

Yes, you're right. We'll Push Docker Images Based on release tags only, that way we'll have a different image for each release but as @agriyakhetarpal mentioned we're discussing which Image should we push for release to keep it single and simple for each release (i.e. We're mostly in favor of image with `IDAKLU` or `ALL` solver), We'd be happy to have your opinion on this @Saransh-cpp

Reply:
Yes, we're also discussing reducing the number of images we build at this time – the reason is that the `latest` image without any solvers is just a simple git clone command and doesn't make sense to include, the `jax` image can be superseded by `pip install pybamm[jax]` where wheels are currently available for all platforms, which makes the installation a matter of seconds. The images where there are any compilation steps present are `odes`, `idaklu` and subsequently `all` – all of them use the same SUNDIALS that has been compiled in `.local/`. My opinion is towards just pushing the `all` image to accommodate all solvers ehile pushing no other images. We run the image builds on commits to the main branch and the arm64 image workflow runs for ~40 mins due to QEMU emulation, so we are not constrained about time or size either. @arjxn-py, could you open a new issue about this?

Reply:
Also, we've been mixing `conda` and `virtualenv` environments in the image – while that does work without issues now, it's not really a good idea to do so, even in late 2023. I would suggest switching to using a manylinux2014-compliant base image (i.e., with `glibc`) in the not-so-long term; it will essentially be the process of containerising PyBaMM from the ground up once again, but the added experience in this area for all of us in the past months would mean it can be done pretty quickly.

Reply:
> could you open a new issue about this?

Sure, I'm up for making the `all` image as standard one if we do not consider size. Although, this is also to be noted that `latest`, `jax` & `odes` image with `all` also manages the environment & installation from source which I feel is different than `pip install pybamm[extras]`. Happy to open a new Issue about the same and have a discussion there 🙂

Reply:
Yeah ultimately it doesn't make sense for the `jax` image to have SUNDIALS pre-installed. The [odes] extra is the only one where we don't have wheels on PyPI so we have to compile from source – perhaps they are not publishing because of copyleft licensing issues...

Reply:
> won't this overwrite the "latest" (develop) image on dockerhub with the "release" image, and then as soon as a new commit is pushed to develop, the "release" image will be overwritten by a new "latest" image?

Guess I missed this while answering previously, Sorry.
No, this wont overwrite the previous release image as two different release images would have two different tags and as soon as tags are different the images are distinct. Maybe you can notice [here](https://hub.docker.com/_/python/tags) how there are different tags and different images.